pairinteraction/CMakeLists.txt

```diff
@@ -113,10 +113,88 @@ set_target_properties(wignerSymbols PROPERTIES POSITION_INDEPENDENT_CODE ON)
 target_link_libraries(pireal PRIVATE wignerSymbols)
 target_link_libraries(picomplex PRIVATE wignerSymbols)
 
-# Include Eigen
+set(MKL_ARCH "intel64" CACHE STRING "MKL architecture")
+set(MKL_LINK "dynamic" CACHE STRING "MKL linkage")
+set(MKL_THREADING "sequential" CACHE STRING "MKL threading backend")
+set(MKL_INTERFACE "lp64" CACHE STRING "MKL interface")
+find_package(MKL CONFIG)
+find_package(PkgConfig)
+if(MKL_FOUND)
+  target_link_libraries(pireal PUBLIC MKL::MKL)
+  target_link_libraries(picomplex PUBLIC MKL::MKL)
+  target_compile_definitions(pireal PUBLIC WITH_INTEL_MKL)
+  target_compile_definitions(picomplex PUBLIC WITH_INTEL_MKL)
+elseif(PKG_CONFIG_FOUND)
+  if(MKL_LINK STREQUAL "sdl")
+    pkg_check_modules(MKL mkl-sdl IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "sequential")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-seq IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "intel_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-iomp IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "gnu_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-gomp IMPORTED_TARGET)
+  elseif(MKL_THREADING STREQUAL "tbb_thread")
+    pkg_check_modules(MKL mkl-${MKL_LINK}-${MKL_INTERFACE}-tbb IMPORTED_TARGET)
+  endif()
+  if(MKL_FOUND)
+    target_link_libraries(pireal PUBLIC PkgConfig::MKL)
+    target_link_libraries(picomplex PUBLIC PkgConfig::MKL)
+    target_compile_definitions(pireal PUBLIC WITH_INTEL_MKL)
+    target_compile_definitions(picomplex PUBLIC WITH_INTEL_MKL)
+  endif()
+endif()
+
+find_package(BLAS)
```

Comment:
If mkl is found, the result of `find_package(BLAS/LAPACK)` will be meaningless - even if it did not return that the packages are found, we would still use intel's blas and lapacke. Thus, I would only search for the packages if mkl is not found.

Reply:
I don't like so many levels of nesting and it takes away in terms of flexibilty. Now I cannot link to different LAPACK implementations for different targets, e.g. for the benchmarks in #139 where I would like one benchmark that links to OpenBLAS and another that links to MKL.

Reply:
I don't see how it takes away flexibility. If MKL is found, the results of find_package(BLAS/LAPACK) were never used anyway if I understood the code correctly. The flexibility to link against OpenBLAS or MKL did not exist because of `if(WITH_LAPACK AND LAPACK_FOUND AND NOT MKL_FOUND)`. However,  I agree that, in principle, one could write the code in such a way that one has this flexibility and that this would be a better solution than my proposal.

Reply:
We could check whether `BLA_VENDOR` is set, and if yes, take the lapacke found by `find_package(BLAS/LAPACK)` instead of intel MKL's lapacke. If you agree that this would be a good solution, I can also look into it.